packages/host/app/modifiers/code-block.ts

```diff
@@ -0,0 +1,172 @@
+import { registerDestructor } from '@ember/destroyable';
+
+import { isTesting } from '@embroider/macros';
+
+import Modifier from 'ember-modifier';
+
+import * as MonacoSDK from 'monaco-editor';
+
+import { MonacoEditorOptions } from './monaco';
+
+import '@cardstack/requirejs-monaco-ember-polyfill';
+
+interface Signature {
+  Args: {
+    Named: {
+      codeBlockSelector: string;
+      languageAttr: string;
+      monacoSDK: typeof MonacoSDK;
+      editorDisplayOptions?: MonacoEditorOptions;
+      index: number;
+    };
+  };
+}
+
+export default class CodeBlock extends Modifier<Signature> {
+  private monacoState: {
+    editor: MonacoSDK.editor.IStandaloneCodeEditor;
+    model: MonacoSDK.editor.ITextModel;
+    language: string | undefined;
+  }[] = [];
+  modify(
+    element: HTMLElement,
+    _positional: [],
+    {
+      codeBlockSelector,
+      languageAttr,
+      monacoSDK,
+      editorDisplayOptions,
+    }: Signature['Args']['Named'],
+  ) {
+    let codeBlocks = element.querySelectorAll(codeBlockSelector);
+    if (!codeBlocks || codeBlocks.length === 0) {
+      return;
+    }
+
+    for (let [index, codeBlockNode] of [...codeBlocks].entries()) {
+      let codeBlock = codeBlockNode as HTMLElement;
+      let id = codeBlock.id;
+      if (!id) {
+        continue;
+      }
+      // note that since the localstorage item was set from runtime-common there
+      // was no ember-window-mock available, so we can't use window mock here
+      let maybeContent = window.localStorage.getItem(id);
```

Comment:
It’s worrisome to me that [this custom lint rule](https://github.com/cardstack/boxel/blob/3b2b6f8dcd5febe78c570be47e434c1836393c37/packages/eslint-plugin-window-mock/mock-window-only.js) isn’t triggering 😳